core-patch/src/test/scala/eu/ostrzyciel/jelly/core/patch/PatchDecoderSpec.scala

```diff
@@ -0,0 +1,354 @@
+          .withStatementType(stType)
+          .withStreamType(PatchStreamType.FLAT)
+        ),
+        RdfPatchRow(null, 100),
+      ))
+      val decoder = decFactory(null, None)
+      val e = intercept[RdfProtoDeserializationError] {
+        decoder.ingestFrame(input)
+      }
+      e.getMessage should include("Row kind is not set or unknown: 100")
+    }
+  }
+
+  "TriplesDecoder" should {
+    "decode triples (1)" in {
+      val input = PatchTestCases.Triples1.encodedFull(
+        JellyPatchOptions.smallStrict
+          .withStatementType(PatchStatementType.TRIPLES)
+          .withStreamType(PatchStreamType.FLAT),
+        10_000
+      ).head
+      val out = PatchCollector()
+      val decoder = MockPatchConverterFactory.triplesDecoder(out, None)
+      decoder.ingestFrame(input)
+      out.statements.result() shouldBe PatchTestCases.Triples1.mrl
+    }
+
+    "decode triples (2) with namespace declarations" in {
+      val input = PatchTestCases.Triples2NsDecl.encodedFull(
+        JellyPatchOptions.smallStrict
+          .withStatementType(PatchStatementType.TRIPLES)
+          .withStreamType(PatchStreamType.FLAT),
+        10_000
+      ).head
+      val out = PatchCollector()
+      val decoder = MockPatchConverterFactory.triplesDecoder(out, None)
+      decoder.ingestFrame(input)
+      out.statements.result() shouldBe PatchTestCases.Triples2NsDecl.mrl
+    }
+
+    "ignore graph names in quads" in {
+      val input = RdfPatchFrame(Seq(
+        RdfPatchRow.ofOptions(JellyPatchOptions.smallStrict
+          .withStatementType(PatchStatementType.TRIPLES)
+          .withStreamType(PatchStreamType.FLAT)
+        ),
+        RdfPatchRow.ofStatementAdd(RdfQuad(
+          RdfTerm.Bnode("b1"), RdfTerm.Bnode("b1"), RdfTerm.Bnode("b1"), RdfTerm.Bnode("b1")
+        )),
+        RdfPatchRow.ofStatementDelete(RdfQuad(
+          RdfTerm.Bnode("b1"), RdfTerm.Bnode("b1"), RdfTerm.Bnode("b1"), RdfTerm.Bnode("b1")
+        )),
+      ))
+      val out = PatchCollector()
+      val decoder = MockPatchConverterFactory.triplesDecoder(out, None)
+      decoder.ingestFrame(input)
+      out.statements.size should be (2)
+      out.statements should be (Seq(
+        Mpl.Add(Mrl.Triple(
+          Mrl.BlankNode("b1"), Mrl.BlankNode("b1"), Mrl.BlankNode("b1")
+        )),
+        Mpl.Delete(Mrl.Triple(
+          Mrl.BlankNode("b1"), Mrl.BlankNode("b1"), Mrl.BlankNode("b1")
+        )),
+      ))
+    }
+
+    "not accept a stream with statement type QUADS" in {
+      val input = RdfPatchFrame(Seq(
+        RdfPatchRow.ofOptions(JellyPatchOptions.smallStrict
+          .withStatementType(PatchStatementType.QUADS)
+          .withStreamType(PatchStreamType.FLAT)
+        ),
+      ))
+      val decoder = MockPatchConverterFactory.triplesDecoder(null, None)
+      val e = intercept[RdfProtoDeserializationError] {
+        decoder.ingestFrame(input)
+      }
+      e.getMessage should include("Incoming stream with statement type")
```

Comment:
eeee, matching error message with strings in this way, didn't we want to make it nicer

Reply:
That would require a complete rewrite of exception handling in Jelly. You sure?

Reply:
eh, let's make it a suggestion for later